### ISSUE TITLE
Sauvegarde des fournisseurs au niveau du compte utilisateur

### ISSUE THREAD

**Issue by lecault** [geobretagne/mviewerstudio#53]
Bonjour,

Il serait bien d'avoir la possibilité de sauvegarder au niveau du compte de l'utilisateur ses fournisseurs de données.

Actuellement, il y a :
- un fichier de conf global où les serveurs sont listés (config-sample.json)
- la possibilité d'ajouter un serveur mais qui n'est pas conservé après la fermeture de session.

![image](https://user-images.githubusercontent.com/5269430/74350433-031f7000-4db6-11ea-8860-dcfd96e306eb.png)

Ici le serveur CD35. De même, il serait bien d'avoir la possibilité de le supprimer.

A voir où positionner ce stockage, variable selon la méthode d'authentification.

**Comment by bchartier** [comment]
A stocker dans le `localStorage` du navigateur ?

**Comment by spelhate** [comment]
C'est une idée. En attendant de disposer d'un backend.

**Comment by RemiDesgrange** [comment]
@spelhate on dispose déjà d'un backend non ? Le backend étant de plus en plus complexe. On pourrait utiliser SQLite (ce qui fonctionne toujours, même avec le provider le plus cheap de la planete).

https://www.rdegges.com/2018/please-stop-using-local-storage/ : c'est un des articles (certes assez "opiniatre") si vous voulez faire ce genre de choses, plutôt utiliser IndexedDB.

**Comment by bchartier** [comment]
> https://www.rdegges.com/2018/please-stop-using-local-storage/ : c'est un des articles (certes assez "opiniatre") si vous voulez faire ce genre de choses, plutôt utiliser IndexedDB.

Pour résumer, l'article dit d'arrêter de stocker des informations de session, et en particulier des informations sensibles, dans le localstorage surtout pour des questions de sécurité. Il ne jette pas le bébé avec l'eau du bain.

**Comment by bchartier** [comment]
Je corrige, l'essence à retenir de l'article est là : 

> Unless you need to store publicly available information that:
> - Is not at all sensitive
> - Doesn’t need to be used in an ultra high performance app
> - Isn’t larger than 5MB
> - Consists of purely string data
> … don’t use local storage! Use the right tool for the job.

**Comment by spelhate** [comment]
> @spelhate on dispose déjà d'un backend non ? Le backend étant de plus en plus complexe. On pourrait utiliser SQLite (ce qui fonctionne toujours, même avec le provider le plus cheap de la planete).

Oui il existe bien un fragment de backend limité à l'enregistrement et au listing de fichiers de configuration.
Pour muscler le backend actuel, j'aime beaucoup l'idée d'utiliser SQLite qui ne nécessite pas d'installation particulière.

**Comment by spelhate** [comment]
> Je corrige, l'essence à retenir de l'article est là :
> 
> > Unless you need to store publicly available information that:
> > 
> > * Is not at all sensitive
> > * Doesn’t need to be used in an ultra high performance app
> > * Isn’t larger than 5MB
> > * Consists of purely string data
> >   … don’t use local storage! Use the right tool for the job.

Et donc c'est tout bon pour stocker 2 ou 3 infos liées à la session de l'utilisateur - comme un fournisseur de données - pour en revenir au sujet.

**Comment by spelhate** [comment]
Si je synthétise, on a 2 possibilités pour stocker des url de services de données :
 

- le localstorage
- une base de données type SQLite

L'avantage du localstorage est d'être plus facile à mettre en oeuvre (tout se passe côté client).
L'avantage de la base de données est d'offrir d'avantage de possibilités pour les évolutions futures mais nécessite le développement d'un backend adapté. Tout dépend du temps que vous pouvez y consacrer.
Les deux solutions envisagées me conviennent pour cette évolution.

**Comment by bchartier** [comment]
> > Je corrige, l'essence à retenir de l'article est là :
> > > Unless you need to store publicly available information that:
> > > 
> > > * Is not at all sensitive
> > > * Doesn’t need to be used in an ultra high performance app
> > > * Isn’t larger than 5MB
> > > * Consists of purely string data
> > >   … don’t use local storage! Use the right tool for the job.
> 
> Et donc c'est tout bon pour stocker 2 ou 3 infos liées à la session de l'utilisateur - comme un fournisseur de données - pour en revenir au sujet.

Oui, mais là on parle d'informations sensibles (sensitive). Aucune information personnelle, aucun mot de passe, aucune clef... Le contenu des fichiers produits par mviewerstudio me paraît nettement plus sensible par exemple (identité de l'auteur, zone géographique d'intérêt, liste complète des couches...). Mais cela ne me dérange pas de penser à autre chose que LocalStorage. C'est juste que les autres solutions seront plus complexes à gérer.
